### PR TITLE
585 hathi links

### PR DESCRIPTION
**Pull request by banukutlu**
Issue #585

### CODE REVIEW

**Comment by banukutlu** [comment]
@cdmo Would you mind doing a quick review for this? It is still WIP but I'd like to get your thoughts/feedback before moving on with the rest.

**Comment by cdmo** [comment]
Sure thing

**Comment by cdmo** [comment]
Or I guess you're not planning on adding the Solr field `hathi_struct` to the catalog controller? I think I was expecting something similar to the way we process other fields in Solr, and `full_links_struct` for example.

File: app/controllers/catalog_controller.rb [hunk 1]
```diff
@@ -181,6 +181,7 @@ def index
     config.add_index_field 'publication_display_ssm', label: 'Publication Statement', sr_only: true
     config.add_index_field 'edition_display_ssm', label: 'Edition', sr_only: true
     config.add_index_field 'full_links_struct', label: 'Access Online', helper_method: :generic_link
+    config.add_index_field 'hathitrust_struct', label: 'HathiTrust Access', helper_method: :hathi_link, hide_label: true
```
Comment (cdmo) [inline app/controllers/catalog_controller.rb:184]: Can you use the `sr_only` flag instead? I replaced the "hide label" functionality in this #580

Comment (banukutlu) [reply app/controllers/catalog_controller.rb]: just tried with `sr_only` and it did not hide the label, is there sth else to do with that flag to hide the label?

Comment (cdmo) [reply app/controllers/catalog_controller.rb]: I see, ok. Well it should be working fine for the index view at least. The sr_only logic would have to be added to the show partials too.

Leaving the label out creates an accessibility problem and a semantic web problem. You'd have a `dl` child item with only a `dd` and no `dt` in cases of "hiding the label."

Comment (banukutlu) [reply app/controllers/catalog_controller.rb]: ahh ok, I will add that logic to show as well then use it.

File: app/views/catalog/_show_default.html.erb [hunk 2]
```diff
@@ -2,7 +2,7 @@
 <%# default partial to display solr document fields in catalog show view -%>
 <dl>
   <% doc_presenter.fields_to_render.each do |field_name, field| -%>
-    <dt class="blacklight-<%= field_name.parameterize %>-label"><%= render_document_show_field_label document, field: field_name %></dt>
-    <dd class="blacklight-<%= field_name.parameterize %> ml-5"><%= doc_presenter.field_value field %></dd>
+      <dt class="blacklight-<%= field_name.parameterize %>-label"><%= render_document_show_field_label document, field: field_name %></dt>
+      <dd class="blacklight-<%= field_name.parameterize %> ml-5"><%= doc_presenter.field_value field %></dd>
```
Comment (banukutlu) [inline app/views/catalog/_show_default.html.erb:6]:
```suggestion
    <dt class="blacklight-<%= field_name.parameterize %>-label"><%= render_document_show_field_label document, field: field_name %></dt>
    <dd class="blacklight-<%= field_name.parameterize %> ml-5"><%= doc_presenter.field_value field %></dd>
```

File: solr/conf/solrconfig.xml [hunk 3]
```diff
@@ -224,7 +224,8 @@
          full_links_struct,
          isbn_valid_ssm,
          oclc_number_ssim,
-         lccn_ssim
+         lccn_ssim,
+         hathitrust_struct,
```
Comment (banukutlu) [inline solr/conf/solrconfig.xml:228]:
```suggestion
         hathitrust_struct
```

File: app/views/catalog/_show_default.html.erb (hunk 2 above)
Comment (cdmo) [reply app/views/catalog/_show_default.html.erb]: :)

**Comment by cdmo** [comment]
Fleshed out `ht_link_text` as per https://github.com/psu-libraries/psulib_blacklight/pull/597#issuecomment-651769758 in f1ab9e1

File: app/views/catalog/_show_top_fields.html.erb [hunk 4]
```diff
@@ -3,8 +3,14 @@
...
   <% blacklight_config.show_fields.each do |field_name, field| -%>
     <% if field.top_field && !document[field_name].nil? %>
-        <dt class="blacklight-<%= field_name.parameterize %>-label col-md-3"><%= render_document_show_field_label document, field: field_name %></dt>
-        <dd class="blacklight-<%= field_name.parameterize %> col-md-9"><%= doc_presenter.field_value field %></dd>
+      <dt class="blacklight-<%= field_name.parameterize %>-label col-md-3
+          <%= 'text-muted' unless field[:sr_only] -%>
+          <%= 'sr-only' if field[:sr_only] -%>">
+        <%= render_document_show_field_label document, field: field_name %>
+      </dt>
+      <dd class="blacklight-<%= field_name.parameterize %> col-md-9"><%= doc_presenter.field_value field %></dd>
```
Comment (cdmo) [inline app/views/catalog/_show_top_fields.html.erb:11]: We don't have any `sr_only` fields as show fields, but, if we do in the future we'll have this all set for it.

**Pull request merged**